### PR TITLE
notebook: fix `cgi.escape` for Python 3.8

### LINKED ISSUE #3015: cgi.escape() no longer exists
### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version 27fd3ab75df3bb79262c4ef0b9bdf2b993092670

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=8, micro=0, releaselevel='final', serial=0)
INFO: os.name: posix
INFO: os.uname(): posix.uname_result(sysname='Linux', nodename='Titan', release='5.4.2-zen1-1-zen', version='#1 ZEN SMP PREEMPT Thu, 05 Dec 2019 12:30:04 +0000', machine='x86_64')
INFO: sys.getwindowsversion(): N/A

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: None

--- check: installed_packages
INFO: installed: tensorboard==2.1.0
INFO: installed: tensorflow==2.1.0rc0
INFO: installed: tensorflow-estimator==2.0.0

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.1.0'

--- check: tensorflow_python_version
2019-12-10 18:48:03.448848: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.2
INFO: tensorflow.__version__: '2.1.0-rc0'
INFO: tensorflow.__git_version__: 'unknown'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'/usr/bin/tensorboard\n'

--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 32>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 32>
Loopback infos: [(<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::1', 0, 0, 0)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('0.0.0.0', 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::', 0, 0, 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): 'localhost'

--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=316518, st_dev=46, st_nlink=2, st_uid=1000, st_gid=1000, st_size=60, st_atime=1576013923, st_mtime=1576013662, st_ctime=1576013662)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['/usr/lib/python3.8/site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==0.8.1
anytree==2.7.3
appdirs==1.4.3
asn1crypto==1.2.0
astor==0.8.0
attrs==19.3.0
autobahn==19.11.1
Automat==0.8.0
backcall==0.1.0
bcrypt==3.1.7
Beaker==1.11.0
beautifulsoup4==4.8.1
black==19.10b0
bleach==3.1.0
blosc==1.8.1
Bottleneck==1.3.1
Brlapi==0.7.0
btrfsutil==1.1.1
CacheControl==0.12.5
cached-property==1.5.1
cachetools==3.1.1
ceph-volume==1.0.0
cephfs==2.0.0
cephfs-shell==0.0.1
cffi==1.13.2
chardet==3.0.4
chrome-gnome-shell==0.0.0
Click==7.0
colorama==0.4.1
colorlog==4.0.2
constantly==15.1.0
cryptography==2.8
cycler==0.10.0
Cython==0.29.14
decorator==4.4.1
defusedxml==0.6.0
distlib==0.3.0
distro==1.4.0
docker==4.1.0
docker-compose==1.25.0
docker-pycreds==0.4.0
dockerpty==0.4.1
docopt==0.6.2
entrypoints==0.3
et-xmlfile==1.0.1
evdev==1.2.0
fail2ban==0.10.4
flake8==3.7.9
Flask-Compress==1.4.0
gast==0.2.2
gevent==1.4.0
gmpy2==2.1.0b1
google-auth==1.8.1
google-auth-oauthlib==0.4.1
google-pasta==0.1.7
greenlet==0.4.15
grpcio==1.25.0
h5py==2.10.0
html5lib==1.0.1
humanfriendly==4.18
hyperlink==19.0.0
idna==2.8
imbalanced-learn==0.6.1
imblearn==0.0
importlib-metadata==0.23
incremental==17.5.0
iotop==0.6
ipykernel==5.1.3
ipython==7.10.1
ipython-genutils==0.1.0
ipywidgets==7.5.1
isc==2.0
itsdangerous==1.1.0
jdcal==1.4.1
jedi==0.15.1
Jinja2==2.10.3
joblib==0.14.0
json5==0.8.5
jsonschema==3.2.0
jupyter-client==5.3.4
jupyter-console==6.0.0
jupyter-core==4.6.1
jupyterlab==1.2.3
jupyterlab-server==1.0.6
Keras==2.3.1
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
kiwisolver==1.1.0
lensfun==0.3.2
libvirt-python==5.8.0
lightdm-gtk-greeter-settings==1.2.2
lit==0.9.0.dev0
lockfile==0.12.2
louis==3.12.0
lutris==0.5.4
lxml==4.4.2
Mako==1.0.14
mallard-ducktype==1.0.2
Markdown==3.1.1
MarkupSafe==1.1.1
matplotlib==3.1.1
mccabe==0.6.1
meson==0.52.1
mistune==0.8.4
more-itertools==7.2.0
MouseInfo==0.1.2
mpi4py==3.0.3
mpmath==1.1.0
msgpack==0.6.2
nbconvert==5.6.1
nbformat==4.4.0
nltk==3.4.5
nose==1.3.7
notebook==6.0.2
numexpr==2.7.0
numpy==1.17.4
oauthlib==3.1.0
openpyxl==3.0.1
opt-einsum==3.1.0
ordered-set==3.1.1
packaging==19.2
pandas==0.25.3
pandas-datareader==0.8.1
pandocfilters==1.4.2
paramiko==2.6.0
parso==0.5.1
pathspec==0.6.0
pdfminer.six==20191107
pep517==0.6.0
pexpect==4.7.0
pickleshare==0.7.5
Pillow==6.2.1
pip==19.2.3
ply==3.11
progress==1.5
prometheus-client==0.7.1
prompt-toolkit==3.0.2
protobuf==3.10.1
psutil==5.6.7
ptyprocess==0.6.0
pwquality==1.4.2
pyalpm==0.9.0
pyasn1==0.4.8
pyasn1-modules==0.2.7
PyAutoGUI==0.9.48
pycairo==1.18.2
pycodestyle==2.5.0
pycparser==2.19
pycryptodome==3.9.4
pycuda==2019.1.2
pycurl==7.43.0.3
pyflakes==2.1.1
PyGetWindow==0.0.8
Pygments==2.5.2
PyGObject==3.34.0
PyHamcrest==1.9.0
pyinotify==0.9.6
PyMsgBox==1.0.7
PyNaCl==1.3.0
pyopencl==2019.1.2
pyparsing==2.4.5
pyperclip==1.7.0
Pypubsub==4.0.3
PyQt5==5.13.2
PyQt5-sip==4.19.19
PyRect==0.1.4
pyrsistent==0.15.6
PyScreeze==0.1.25
pyserial==3.4
python-dateutil==2.8.1
python-engineio==3.10.0
python-libtorrent==1.1.13
python-pam==1.8.4
python-socketio==4.3.1
python-xapp==1.8.1
python3-xlib==0.15
pytoml==0.1.21
pytools==2019.1.1
PyTweening==1.0.3
pytz==2019.3
pyxdg==0.26
PyYAML==5.1.2
pyzmq==18.1.1
rados==2.0.0
rbd==2.0.0
Reflector==2019.10.15.3.19.34
regex==2019.11.1
requests==2.22.0
requests-oauthlib==1.3.0
retrying==1.3.3
rgw==2.0.0
rsa==4.0
scikit-learn==0.22
scikit-plot==0.3.7
scipy==1.3.1
seaborn==0.9.0
selenium==3.141.0
Send2Trash==1.5.0
setproctitle==1.1.10
setuptools==41.6.0
six==1.13.0
smbus==1.1
sortedcontainers==2.1.0
soupsieve==1.9.5
SQLAlchemy==1.3.11
sympy==1.4
systemd-python==234
team==1.0
tensorboard==2.1.0
tensorflow==2.1.0rc0
tensorflow-estimator==2.0.0
tensorflow-serving-api==2.0.0
tensorflow-serving-api-gpu==2.0.0
termcolor==1.1.0
terminado==0.8.3
testpath==0.4.4
texttable==1.6.2
tinycss==0.4
tldr==0.5
toml==0.10.0
tornado==6.0.3
tqdm==4.38.0
traitlets==4.3.3
Twisted==19.10.0
txaio==18.8.1
typed-ast==1.4.0
urllib3==1.25.7
wcwidth==0.1.7
webencodings==0.5.1
websocket-client==0.56.0
websockets==8.1
Werkzeug==0.16.0
wheel==0.33.6
widgetsnbextension==3.5.1
wrapt==1.11.2
wsaccel==0.6.2
wxPython==4.0.7.post2
xlrd==1.2.0
XlsxWriter==1.2.6
xlwt==1.3.0
youtube-dl==2019.11.28
zipp==0.6.0
zope.interface==4.7.1

``````

</details>

## Issue description

The `escape()` function has been removed from the `cgi` package in Python 3.8, as seen [here](https://docs.python.org/3.8/whatsnew/3.8.html#api-and-feature-removals). The suggestion is to use `html.escape()` instead, but since it's only available from Python 3.2 onwards, a hack might be needed to get this working, as seen in SciTools/cartopy#1395.

### PR DESCRIPTION
**Pull request by igormp**
Fixes #3015

Simply imports `html` instead of `cgi` if our Python version is 3.8 or newer, keeps `cgi` otherwise in order to maintain backwards compat.

Importing `html` as `cgi` is kinda ugly IMO, but I'm not really sure on how to fix it without having another if condition when calling `escape()`.

Another option would be to do `from html import escape`, but it seems that it'd break the code style currently going on for external libraries.

### CODE REVIEW

**Comment by wchargin** [comment]
That said, we expect to drop Python 2 support quite soon, at which point
we’ll not need to support Python versions earlier than 3.2, either, so
this isn’t a big deal either way. :-)

**Comment by igormp** [comment]
From what I've read, Python 3.8 support will only be available with tf 2.2. Will it drop Python 2 already? If so, we could simply import `html` and forgo `cgi`.

**Comment by wchargin** [comment]
We need to support Python 2 until at least the end of the year (i.e.,
the next 21 days), including in `master`, and we’ve [pledged][1] to drop
Python 2 support at some time in 2020. I’d expect that to be early in
the year, but I can’t say for sure, and I’m likewise not sure what
TensorFlow’s plans are for supporting Python 3.8.

So it’s okay with me if the version guard is a bit awkward or ugly, but
we do need to support Python 2 for a little longer, so we can’t just
forgo `cgi`.

[1]: https://python3statement.org/

**Comment by igormp** [comment]
No problems, I've updated the PR. 

I hope someday we'll be able to get rid of Python2's ghost.

**Comment by wchargin** [comment]
Thanks! Tested in Python 2 and 3; works on my machine.

<details>
<summary>Commit message (note to self)</summary>

``````
The deprecated `cgi.escape` function is removed in Python 3.8, in favor
of `html.escape`, which exists in Python 3.2+. This commit uses updates
the notebook integration to use the appropriate endpoint on all Pythons.

Tested:
Run `bazel run //tensorboard/pip_package:extract_pip_package`, then
install the built wheels into fresh Python 2 and 3 virtualenvs with
TensorFlow and Jupyter installed. Run

```python
%load_ext tensorboard
%tensorboard --logdir /tmp/whatever

import sys
print("html" in sys.modules)
print("cgi" in sys.modules)
```

in each, and note that in each case the notebook extensions work, while
the imported modules are different in the two environments: `False True`
in Python 3, `True False` in Python 2.
``````

</details>
